cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/StreamHandler.java

```diff
@@ -347,25 +337,21 @@ private StreamConfig getAndValidateConfig(StreamConfig currConfig, HttpRequest r
       return null;
     }
 
-    // if no ttl is given, use the existing ttl.
-    Long newTTL = properties.getTTL();
-    if (newTTL == null) {
-      newTTL = currConfig.getTTL();
-    } else {
-      if (newTTL < 0) {
+    // Validate ttl
+    Long ttl = properties.getTTL();
+    if (ttl != null) {
+      if (ttl < 0) {
         responder.sendString(HttpResponseStatus.BAD_REQUEST, "TTL value should be positive.");
         return null;
       }
       // TTL in the REST API is in seconds. Convert it to ms for the config.
-      newTTL = TimeUnit.SECONDS.toMillis(newTTL);
+      ttl = TimeUnit.SECONDS.toMillis(ttl);
```

Comment:
This is from the old code. Probably that's the place of inconsistency about converting the unit of ttl. Will take a look into that.